### PR TITLE
Clean up requirements and expand developer guide

### PR DESCRIPTION
**Pull request by GabrielBianconi**
> [!IMPORTANT]
> Clean up requirements, update workflows, and expand developer guide for improved setup and testing.
> 
>   - **Workflows**:
>     - Remove `dashboard` branch from `general.yml` and `pr.yml`.
>     - Update `general.yml` to specify `ruff==0.8.4` and add mypy checks for `clients/python` and `clients/openai-python`.
>     - Remove dashboard test services from `general.yml`.
>     - Add health check for gateway in `merge-queue.yml`.
>     - Add detailed Python client test steps in `merge-queue.yml`.
>   - **Requirements**:
>     - Split `clients/requirements.txt` into `clients/python/requirements.txt` and `clients/openai-python/requirements.txt`.
>     - Update dependencies in `requirements.txt` files for `clients/python` and `clients/openai-python`.
>     - Add `requirements.in` files for `clients/python` and `clients/openai-python`.
>   - **Developer Guide**:
>     - Add setup instructions for Rust, Docker, and Python in `CONTRIBUTING.md`.
>     - Include detailed testing instructions for Rust and Python in `CONTRIBUTING.md`.
>   - **Misc**:
>     - Add type hint to `test_double_feedback_query()` in `test_clickhouse_queries.py`.
>     - Update `pre-commit-config.yaml` to use `ruff` version `0.8.4`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=tensorzero%2Ftensorzero&utm_source=github&utm_medium=referral)<sup> for db40ee4502d9f4ebc365d4d886e60d31346b0ab2. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .github/workflows/general.yml [hunk 1]
```diff
@@ -92,16 +99,3 @@ jobs:
...
-        run: |
-          echo "OPENAI_API_KEY=${{secrets.OPENAI_API_KEY}}" >> .env
-          echo "FIREWORKS_API_KEY=${{secrets.FIREWORKS_API_KEY}}" >> .env
-          echo "FIREWORKS_ACCOUNT_ID=${{secrets.FIREWORKS_ACCOUNT_ID}}" >> .env
-          echo "CLICKHOUSE_URL=http://localhost:8123/tensorzero" >> .env
-          docker compose -f fixtures/docker-compose.yml up -d
-
-      - name: Run dashboard tests
-        working-directory: dashboard
-        run: npm test
```
Comment (GabrielBianconi) [inline .github/workflows/general.yml:107]: We don't want API calls in this file

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -4,6 +4,17 @@
 >
 > **This guide is for developers planning to contribute to TensorZero, not for developers planning to use TensorZero.**
```
[the remark this answers is not in the document]
Comment (virajmehta) [reply DEVELOPER_GUIDE.md:5]: or add to CONTRIBUTING.md with a table of contents? or at minimum link to it from CONTRIBUTING.md?

**Review by virajmehta — CHANGES_REQUESTED** [review]
put the npm tests back for now, couple comments otherwise that aren't hard requirements to merge

File: DEVELOPER_GUIDE.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (GabrielBianconi) [reply DEVELOPER_GUIDE.md]: merged

**Review by virajmehta — APPROVED** [review]
LGTM!

**Pull request merged**